app/views/layouts/_sidebox.html.erb

```diff
@@ -64,12 +64,14 @@
           </div>
         <% end %>
 
-        <% if current_user.payments.current_camp_payments.present? %>
-          <div class="sidebox_element">
-            <%= link_to "Application Fee",
-                        all_payments_path %>
-               - paid
-          </div>
+        <% if @curr_enrollment&.application_fee_required %>
+          <% if current_user.payments.current_camp_payments.present? %>
```

Comment:
I think, status1_current_camp_payments scope should be used in line 68 instead of current_camp_payments

Reply:
makes senses